lxc/copy.go

```diff
@@ -269,23 +285,41 @@ func (c *cmdCopy) copyInstance(conf *config.Config, sourceResource string, destR
 			entry.Profiles = []string{}
 		}
 
-		// Allow setting additional config keys
-		for key, value := range configMap {
-			entry.Config[key] = value
+		// Check to see if any of the overridden devices are for devices that are not yet defined in the
+		// local devices (and thus maybe expected to be coming from profiles).
+		needProfileExpansion := false
+		for deviceName := range deviceOverrides {
+			_, isLocalDevice := entry.Devices[deviceName]
+			if !isLocalDevice {
+				needProfileExpansion = true
+				break
+			}
 		}
 
-		// Allow setting device overrides
-		for k, m := range deviceMap {
-			if entry.Devices[k] == nil {
-				entry.Devices[k] = m
-				continue
-			}
+		profileDevices := make(map[string]map[string]string)
 
-			for key, value := range m {
-				entry.Devices[k][key] = value
+		// If there are device overrides that are expected to be applied to profile devices then perform
+		// profile expansion.
+		if needProfileExpansion {
+			profileDevices, err = getProfileDevices(dest, entry.Profiles)
+			if err != nil {
+				return err
 			}
 		}
 
+		// Apply device overrides.
+		resultingDevices, err := shared.ApplyDeviceOverrides(entry.Devices, profileDevices, deviceOverrides)
+		if err != nil {
+			return err
+		}
+
+		entry.Devices = resultingDevices
```

Comment:
Can you do the same as above and assign `entry.Devices` directly when calling `shared.ApplyDeviceOverrides`?